pkg/model/options.go

```diff
@@ -200,95 +173,75 @@ func (opts *Options) Validate() error {
 		}
 	}
 
-	// TODO: validate plural strings if provided
-
 	return nil
 }
 
-// GVK returns the group-version-kind information to check against tracked resources in the configuration file
-func (opts *Options) GVK() config.GVK {
-	return config.GVK{
-		Group:          opts.Group,
-		Version:        opts.Version,
-		Kind:           opts.Kind,
-		CRDVersion:     opts.CRDVersion,
-		WebhookVersion: opts.WebhookVersion,
+// QualifiedGroup returns the fully qualified group name with the available information.
+func (opts Options) QualifiedGroup() string {
+	switch "" {
+	case opts.Domain:
+		return opts.Group
+	case opts.Group:
+		return opts.Domain
+	default:
+		return fmt.Sprintf("%s.%s", opts.Group, opts.Domain)
 	}
 }
 
-// safeImport returns a cleaned version of the provided string that can be used for imports
-func (opts *Options) safeImport(unsafe string) string {
-	safe := unsafe
-
-	// Remove dashes and dots
-	safe = strings.Replace(safe, "-", "", -1)
-	safe = strings.Replace(safe, ".", "", -1)
-
-	return safe
+// GVK returns the GVK identifier of a resource.
+func (opts Options) GVK() resource.GVK {
+	return resource.GVK{
+		Group:   opts.QualifiedGroup(),
+		Version: opts.Version,
+		Kind:    opts.Kind,
+	}
 }
 
 // NewResource creates a new resource from the options
-func (opts *Options) NewResource(c *config.Config, doResource bool) *Resource {
-	res := opts.newResource()
-
-	replacer := res.Replacer()
-
-	pkg := replacer.Replace(path.Join(c.Repo, "api", "%[version]"))
-	if c.MultiGroup {
-		if opts.Group != "" {
-			pkg = replacer.Replace(path.Join(c.Repo, "apis", "%[group]", "%[version]"))
-		} else {
-			pkg = replacer.Replace(path.Join(c.Repo, "apis", "%[version]"))
-		}
+func (opts Options) NewResource(c *config.Config) *resource.Resource {
+	// If not provided, compute a plural for for Kind
+	plural := opts.Plural
+	if plural == "" {
+		plural = resource.RegularPlural(opts.Kind)
+	}
+
+	res := &resource.Resource{
+		Group:   opts.Group,
+		Domain:  opts.Domain,
+		Version: opts.Version,
+		Kind:    opts.Kind,
+		Plural:  plural,
+		Path:    resource.LocalPath(c.Repo, opts.Group, opts.Version, c.MultiGroup),
+		API: &resource.API{
+			Version:    opts.CRDVersion,
+			Namespaced: opts.Namespaced,
+		},
```

Comment:
```suggestion
		API: opts.API,
```

Reply:
`model.Options` doesn't have any `API` field, thay are all root-level fields. When we create the resource they are structured properly. Additionally, we want to make sure that `Resource.API` is not nil after `NewResource` is called because it is used in templates like `{{ .Resource.API.Namespaced }}` and that will fail if `API` is nil.

Reply:
I think we ought to add the API/Webhooks in the opts. 
Then, I think we should not change the v2 templates only the v3. 
See; https://github.com/kubernetes-sigs/kubebuilder/pull/1696/files#diff-9913c42e8e81878b81ed66efe824800b277c56005ad491ab5aa09532a93265a1R290-R291

Reply:
`Options` is just a set of variables that contain flags values provided by the users. They are not a resource, they will be used to create one, that's why `Options.API` or `Options.Webhooks` don't make sense semantically.

The change in `v2` templates is discussed in another comment. Let's not mix the different conversations.

Reply:
we can pass an obj as an option since api and webhooks are used in the CLI flags. However, it is fine. Only a nit anyway.